contracts/contracts/harvest/Harvester.sol

```diff
@@ -0,0 +1,376 @@
+// SPDX-License-Identifier: agpl-3.0
+pragma solidity ^0.8.0;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { SafeMath } from "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import "@openzeppelin/contracts/utils/math/Math.sol";
+
+import { StableMath } from "../utils/StableMath.sol";
+import { Initializable } from "../utils/Initializable.sol";
+import { Governable } from "../governance/Governable.sol";
+import { IVault } from "../interfaces/IVault.sol";
+import { IOracle } from "../interfaces/IOracle.sol";
+import { IStrategy } from "../interfaces/IStrategy.sol";
+import { IUniswapV2Router } from "../interfaces/uniswap/IUniswapV2Router02.sol";
+import "../utils/Helpers.sol";
+
+contract Harvester is Initializable, Governable {
+    using SafeERC20 for IERC20;
+    using SafeMath for uint256;
+    using StableMath for uint256;
+
+    event UniswapUpdated(address _address);
+    event SwapTokenAdded(address _address);
+    event SwapTokenRemoved(address _address);
+    event RewardTokenConfigUpdated(
+        address _tokenAddress,
+        uint16 _allowedSlippageBps,
+        uint16 _harvestRewardBps,
+        address _uniswapV2CompatibleAddr,
+        uint256 _liquidationLimit
+    );
+
+    // Tokens that should be swapped for stablecoins
+    address[] public swapTokens;
+
+    // Strategies approved for use by the Vault
+    struct RewardTokenConfig {
+        // Max allowed slippage when swapping reward token for a stablecoin denominated in basis points.
+        uint16 allowedSlippageBps;
+        // Reward when calling a harvest function denominated in basis points.
+        uint16 harvestRewardBps;
+        /* Address of Uniswap V2 compatible exchange (Uniswap V2, SushiSwap).
+         * When this is set to zero address swapping is disabled.
+         */
+        address uniswapV2CompatibleAddr;
+        /* How much token can be sold per one harvest call. If the balance of rewards tokens
+         * exceeds that limit multiple harvest calls are required to harvest all of the tokens.
+         */
+        uint256 liquidationLimit;
+    }
+
+    mapping(address => RewardTokenConfig) public rewardTokenConfigs;
+
+    // Address of Vault
+    address public vaultAddress = address(0);
+    address public usdtAddress = address(0);
+
+    /**
+     * @dev Internal initialize function, to set up initial internal state
+     * @param _vaultAddress Address of the Vault
+     * @param _usdtAddress Address of Tether
+     */
+    function initialize(address _vaultAddress, address _usdtAddress)
+        external
+        onlyGovernor
+        initializer
+    {
+        _initialize(_vaultAddress, _usdtAddress);
+    }
+
+    function _initialize(address _vaultAddress, address _usdtAddress) internal {
+        vaultAddress = _vaultAddress;
+        usdtAddress = _usdtAddress;
```

Comment:
If this was an immutable value set in the constructor, it would save 1600 ish gas per harvest. @tomlinton, do you think it's worth adding a constructor for?

Reply:
Yea I think it is, it's the most gas efficient approach right? For reference Flipper.sol does this @sparrowDom

Reply:
thanks changing to immutable property set inside constructor